### PR TITLE
Using Picasso in Android

### PR DESCRIPTION
**Pull request by BrianaNzivu**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by BrianaNzivu** [comment]
This fixes issue #623 .

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by BrianaNzivu** [comment]
@hectorkambow - No problem. 
Thank you for your patience.

File: articles/using-picasso-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+- Rotation and Transformation of images. 
+- Placeholder and Error images
+- Priority requests
+- Memory and disk Caching.
+- Fading
+- Supports parallel downloading and request cancellation. 
+
+### Advantages of Picasso.
+- Picasso offers high quality loaded images. 
+- Full size caching.
```
Comment (LinusMuema) [inline articles/using-picasso-in-android/index.md:20]: You could give a short detail on what caching does in android especially for images. 

`caching allows for offline capabilities and less resource usage. It ensures the image is not downloaded every time it is needed.`

File: articles/using-picasso-in-android/index.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+### Uses of Picasso
+Picasso has the following features:
+- Image loading.
+- Resizing and Scaling of images. 
+- Center Cropping images.
+- Rotation and Transformation of images. 
+- Placeholder and Error images
+- Priority requests
```
Comment (LinusMuema) [inline articles/using-picasso-in-android/index.md:13]: You could explain these uses more to give the user a richer understanding of the uses.

File: articles/using-picasso-in-android/index.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+	 Picasso.get().load(R.drawable.image).rotate(90f).into(imageView);
+	 break;
+```  
+### h)Targets  
+Targets combine image loading, callbacks, and errors. They return bitmap images. Targets normally use `onBitmapLoaded()`, `onBitmapFailed()` and on `Prepared()` methods. In our `MainActivity.java`, inside the `initializeView` method, add the following lines of code:
+```java
+buttonTarget = (Button) 
+findViewById(R.id.showTarget); buttonTarget.setOnClickListener(this); 
+
+ case R.id.showTarget:
```
Comment (LinusMuema) [inline articles/using-picasso-in-android/index.md:339]: I think this case block goes to the `onClick` method. Make sure you separate these so as not to confuse the reader.
The same applies to other code blocks

**Review by LinusMuema — CHANGES_REQUESTED** [review]
Hi @BrianaNzivu 
This is an awesome article. Really informative.
Just a few suggestions and the article will be ready for final review. 

Side note: The last time I used Picasso in an android application, it had issues with caching images from URLs without image extensions, i.e, if an image URL does not end with `.png`, `.jpg` etc, Picasso won't cache it. I don't know if the issue was solved. If so, well and good. otherwise, inform the users

**Comment by BrianaNzivu** [comment]
Hey @LinusMuema . I have made the changes.
Kindly confirm if everything is OK now.
Regards.

**Comment by LinusMuema** [comment]
Hi @BrianaNzivu 
I have made more changes. Please take a look for future references.

@hectorkambow you could give this article a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
👍  Great tutorial @BrianaNzivu  - it was super informative and well written 🚀  
cc: @LinusMuema

**Pull request merged**